cmd/bucket-handlers.go

```diff
@@ -58,7 +59,7 @@ func enforceBucketPolicy(bucket, action, resource, referer, sourceIP string, que
 	}
 
 	// Fetch bucket policy, if policy is not set return access denied.
-	p, err := objAPI.GetBucketPolicy(bucket)
+	p, err := objAPI.GetBucketPolicy(context.Background(), bucket)
```

Comment:
auth-handlers and PutObject/PutObjectPart need to pass the context to this call.

Reply:
Yes, will be fixed in future